docs/src/charm/howto/upgrade-minor.md

```diff
@@ -0,0 +1,216 @@
+k8s/1*      active    idle   1        3.238.230.3     6443/tcp  Ready
+k8s/2       active    idle   2        34.229.202.243  6443/tcp  Ready
+
+Machine  State    Address         Inst id              Base          AZ          Message
+0        started  54.89.153.117   i-0b6fc845c28864913  ubuntu@22.04  us-east-1f  running
+1        started  3.238.230.3     i-05439714c88bea35f  ubuntu@22.04  us-east-1f  running
+2        started  34.229.202.243  i-07ecf97ed29860334  ubuntu@22.04  us-east-1c  running
+```
+<!-- markdownlint-restore -->
+
+
+## Before you begin
+
+As with all upgrades, there is a possibility that there may be
+unforeseen difficulties. It is highly recommended that you make
+a backup of any important data, including any running workloads.
+For more details on creating backups, see the separate
+[docs on backups][backup-restore].
+
+
+You should also make sure:
+
+* The machine from which you will perform the backup has sufficient
+  internet access to retrieve updated software
+* Your cluster is running normally
+* Your Juju client and controller/models are running the same,
+  stable version of Juju (see the [Juju docs][juju-docs])
+* You read the [Upgrade notes][upgrade-notes] to see if any
+  caveats apply to the versions you are upgrading to/from
+* You read the [Upstream release notes][upstream-notes] for details
+  of Kubernetes deprecation notices and API changes that may impact
+  your workloads
+
+
+It is also important to understand that Kubernetes will only 
+upgrade and if necessary migrate, components relating specifically
+to elements of Kubernetes installed and configured as part of Kubernetes.
+This may not include any customized configuration of Kubernetes,
+or non-built-in generated objects (e.g. storage classes) or deployments which
+rely on deprecated APIs.
+
+## Specific upgrade instructions
+
+### Deciding if an upgrade is available
+
+Juju will contact [charmhub] daily to find new revisions of charms
+deployed in your models. To see if the `k8s` or `k8s-worker` charms 
+can be upgraded, set with the following:
+
+```sh
+juju status --format=json | \
+   jq '.applications | 
+        to_entries[] | {
+           application: .key,
+           "charm-name": .value["charm-name"],
+           "charm-channel": .value["charm-channel"],
+           "charm-rev": .value["charm-rev"],
+           "can-upgrade-to": .value["can-upgrade-to"]
+        }'
+```
+
+This outputs a list of applications in the model:
+* the name of the application (ex. `k8s`)
+* the charm used by the application (ex. `k8s`)
+* the kubernetes channel this charm follows (ex. `1.31/stable`)
+* the current charm revision  (ex. `1001`)
+* the next potential charm revision (ex. `ch:amd64/k8s-1002`)
+
+If the `can-upgrade-to` revision is `null`, you are at the most
+stable release in this channel and you should continue with the
+[Pre Upgrade Check](#the-pre-upgrade-check).
+
+Otherwise, continue with the [Upgrade Patch](upgrade-patch) instructions.
```

Comment:
I am a bit confused to the logic here. Am I right in saying:
 If there can-upgrade-to is null, we are on the latest stable release (up to date in our patches ) in our channel and we can go to to the next channel with the upgrade.
If the can-upgrade-to is not null, we want them to upgrade the patch first.